core/src/main/java/org/mule/runtime/core/context/notification/ServerNotificationManager.java

```diff
@@ -147,24 +158,36 @@ public boolean isListenerRegistered(ServerNotificationListener listener) {
 
   @Override
   public void fireNotification(ServerNotification notification) {
-    if (!disposed.get()) {
+    disposeLock.readLock().lock();
+    try {
+      if (disposed.get()) {
+        logger.warn("Notification not enqueued after ServerNotificationManager disposal: " + notification);
+        return;
+      }
+
       notification.setMuleContext(muleContext);
       if (notification instanceof BlockingServerEvent) {
-        notifyListeners(notification);
+        notifyListeners(notification, (listener, nfn) -> listener.onNotification(nfn));
       } else {
-        try {
-          eventQueue.put(notification);
-        } catch (InterruptedException e) {
-          if (!disposed.get()) {
-            logger.error("Failed to queue notification: " + notification, e);
+        notifyListeners(notification, (listener, nfn) -> {
+          if (CPU_LITE.equals(listener.getProcessingType())) {
```

Comment:
All notifications should be non-blocking.  So there should be a JIRA for this and a TODO here to only use cpuLite pool.  Also, even in the current state where some listeners may be blocking I don't see any need for computation pool here at all.